zebra/zebra_rnh.c

```diff
@@ -1268,73 +1270,181 @@ int zebra_send_rnh_update(struct rnh *rnh, struct zserv *client,
 	return -1;
 }
 
-static void print_nh(struct nexthop *nexthop, struct vty *vty)
+static void print_nh(struct nexthop *nexthop, struct vty *vty,
+		     json_object *json)
 {
 	char buf[BUFSIZ];
 	struct zebra_ns *zns = zebra_ns_lookup(nexthop->vrf_id);
 
 	switch (nexthop->type) {
 	case NEXTHOP_TYPE_IPV4:
 	case NEXTHOP_TYPE_IPV4_IFINDEX:
-		vty_out(vty, " via %pI4", &nexthop->gate.ipv4);
-		if (nexthop->ifindex)
-			vty_out(vty, ", %s",
-				ifindex2ifname_per_ns(zns, nexthop->ifindex));
+		if (json) {
+			json_object_string_addf(json, "ip", "%pI4",
+						inet_ntoa(nexthop->gate.ipv4));
+			if (nexthop->ifindex)
+				json_object_string_add(
+					json, "interface",
+					ifindex2ifname_per_ns(
+						zns, nexthop->ifindex));
+		} else {
+			vty_out(vty, " via %s", inet_ntoa(nexthop->gate.ipv4));
```

Comment:
ditto, inet_ntop.

Reply:
Same here, I believe `vty_out` supports the FRR formatting options. No need for `inet_nto*`.